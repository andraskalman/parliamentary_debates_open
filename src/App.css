@import '~bootswatch/dist/superhero/bootstrap.css';
@import '~react-datetime/css/react-datetime.css';

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.result {
  background: #ffffff1a;
  padding: 0.1em 0.7em;
  padding-bottom: 0.6em;
  margin-bottom: 3.5rem;
}

.result em {
  background: #ff03;
}

.result .meta {
  float: right;
  color: gray;
  font-size: 0.9em;
}

.result h1 {
  font-size: 1.15em;
}

.result .source {
  font-size: 0.9em;
  color: gray;
}

.result .source a {
  text-decoration: underline;
  font-style: italic;
  color: inherit;
}

.rdtPicker {
  background: #2b3e50;
}

.rdtPicker td.rdtDay:hover,
.rdtPicker td.rdtHour:hover,
.rdtPicker td.rdtMinute:hover,
.rdtPicker td.rdtSecond:hover,
.rdtPicker .rdtTimeToggle:hover {
  background: rgba(255, 255, 255, 0.2);
  cursor: pointer;
}

.rdtPicker thead tr:first-child th:hover {
  background: rgba(255, 255, 255, 0.2);
}

.modal-dialog {
  max-width: 900px;
}

.filterTools h2 {
  margin-top: 1em;
}

.filterTools h3 {
  margin-top: 0.5em;
}

.navbar-brand {
  color: rgb(235, 235, 235) !important;
}
