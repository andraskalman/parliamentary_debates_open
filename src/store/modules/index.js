import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import search from './search'
import help from './help'

export default combineReducers({
  routing: routerReducer,
  search,
  help
})
