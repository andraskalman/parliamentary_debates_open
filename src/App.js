import React, {Component} from 'react';
import './App.css';
//import 'bootstrap/dist/css/bootstrap.css';
import 'react-loading-bar/dist/index.css';
import {Container} from 'react-bootstrap';
import Home from './Home';
import {Button, Navbar, Nav} from 'react-bootstrap';
import {toggle} from './store/modules/help';
import {bindActionCreators} from 'redux';

import {connect} from 'react-redux';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faQuestion} from '@fortawesome/free-solid-svg-icons';

class App extends Component {
  UNSAFE_componentWillMount() {
    document.title = 'A magyar országgyűlés felszólalásai 1990 óta';
  }

  render() {
    return (
      <>
        <Navbar bg="dark" expand="lg" sticky="top">
          <Navbar.Brand>Az Országgyűlés felszólalásai 1990 óta</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="mr-auto" />
            <Button onClick={this.props.toggle}>
              <FontAwesomeIcon icon={faQuestion} />
              <span>Súgó mutatása</span>
            </Button>
          </Navbar.Collapse>
        </Navbar>
        <Container style={{marginTop: '2em'}}>
          <Home />
        </Container>
      </>
    );
  }
}

const mapStateToProps = state => ({
  help: state.help,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      toggle,
    },
    dispatch,
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
