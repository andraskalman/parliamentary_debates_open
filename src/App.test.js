import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store, { history } from './store'


it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <div>
        <App />
      </div>
  </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
