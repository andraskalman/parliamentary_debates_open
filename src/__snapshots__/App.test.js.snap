// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          parldata-frontend
        </h1>
      </a>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-3"
      >
        <h2>
          Search
        </h2>
        <input
          className="form-control"
          id={undefined}
          onChange={[Function]}
          type={undefined}
        />
      </div>
      <div
        className="col-sm-9"
      >
        <h2>
          Keresés eredménye
        </h2>
        <p />
      </div>
    </div>
  </div>
</div>
`;
